Stardew Valley not detected on Linux Heroic

# Bug Report

## Summary
Stardew Valley (Linux version) is not detected through Heroic games store. 


## Steps to reproduce
1. Install the native SDV build via Heroic.
2. See it is not detected in the app. 


## What is the expected behaviour?
I can manage SDV


## Other information

[installed.json](https://github.com/user-attachments/files/18167992/installed.json)

[nexusmods.app.main.current.log](https://github.com/user-attachments/files/18168027/nexusmods.app.main.current.log)
[nexusmods.app.main.01.log](https://github.com/user-attachments/files/18168032/nexusmods.app.main.01.log)
[nexusmods.app.main.02.log](https://github.com/user-attachments/files/18168033/nexusmods.app.main.02.log)

Having this issue as well.
Section from logs.
--------------------------------------------------------------------------
00:00:00.477 [ERROR] (NexusMods.StandardGameLocators.HeroicGogLocator) While looking for games: Unable to parse in /home/<USER NAME>/.config/heroic/gog_store/installed.json: `Installed { Platform = linux, Executable = , InstallPath = /home/<USER NAME>/Games/Heroic/Stardew Valley, InstallSize = 676.1 MiB, IsDlc = False, Version = 1.6.15.24357.8705766150, AppName = 1453375253, InstalledDLCs = System.Collections.Generic.List`1[System.Object], Language = en, VersionEtag = , BuildId = , PinnedVersion = False }`:
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Text.Json.JsonElement.GetProperty(String propertyName)
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.Parse(Installed installed, AbsolutePath configPath, IFileSystem fileSystem)
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.ParseInstalledJsonFile(AbsolutePath path, AbsolutePath configPath)+MoveNext()
--------------------------------------------------------------------------

Heroic installed.json file from ~/.config/heroic/gog_store/installed.json
--------------------------------------------------------------------------
File: installed.json
{
    "installed": [
        {
            "platform": "linux",
            "executable": "",
            "install_path": "/home/<USER NAME>/Games/Heroic/Stardew Valley",
            "install_size": "676.1 MiB",
            "is_dlc": false,
            "version": "1.6.15.24357.8705766150",
            "appName": "1453375253",
            "installedDLCs": [],
            "language": "en",
            "versionEtag": "",
            "buildId": "",
            "pinnedVersion": false
        }
    ]
}
--------------------------------------------------------------------------

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -136,7 +136,7 @@
     <PackageVersion Include="BitFaster.Caching" Version="2.5.2" />
     <PackageVersion Include="CliWrap" Version="3.6.7" />
     <PackageVersion Include="DynamicData" Version="9.0.4" />
-    <PackageVersion Include="GameFinder" Version="4.4.0" />
+    <PackageVersion Include="GameFinder" Version="4.5.0" />
     <PackageVersion Include="Humanizer" Version="2.14.1" />
     <PackageVersion Include="ini-parser-netstandard" Version="2.5.2" />
     <PackageVersion Include="Mutagen.Bethesda.Skyrim" Version="0.44.0" />

--- a/src/NexusMods.Games.FileHashes/FileHashesService.cs
+++ b/src/NexusMods.Games.FileHashes/FileHashesService.cs
@@ -300,7 +300,10 @@ public class FileHashesService : IFileHashesService, IDisposable
             foreach (var gogId in locatorMetadata)
             {
                 if (!ulong.TryParse(gogId, out var parsedId))
-                    throw new InvalidOperationException("GOG locator metadata is not a valid ulong");
+                {
+                    _logger.LogWarning("Unable to parse `{Raw}` as ulong", gogId);
+                    return false;
+                }
 
                 var gogBuild = GogBuild.FindByBuildId(Current, BuildId.From(parsedId))
                     .FirstOrDefault();

--- a/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs
+++ b/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs
@@ -1,3 +1,4 @@
+using System.Runtime.InteropServices;
 using GameFinder.Launcher.Heroic;
 using GameFinder.StoreHandlers.GOG;
 using Microsoft.Extensions.DependencyInjection;
@@ -45,6 +46,7 @@ public class HeroicGogLocator : IGameLocator
         {
             if (!_cachedGames.TryGetValue(GOGGameId.From(id), out var found)) continue;
             var fs = found.Path.FileSystem;
+            var gamePath = found.Path;
 
             if (found is HeroicGOGGame heroicGOGGame)
             {
@@ -54,9 +56,15 @@ public class HeroicGogLocator : IGameLocator
                     if (wineData.WinePrefixPath.DirectoryExists())
                         fs = heroicGOGGame.GetWinePrefix()!.CreateOverlayFileSystem(fs);
                 }
+
+                // NOTE(erri120): GOG builds for Linux are whack, the installer Heroic uses is whack,
+                // and this is a complete hack. See comments on https://github.com/Nexus-Mods/NexusMods.App/pull/2653
+                // for details.
+                if (heroicGOGGame.Platform == OSPlatform.Linux)
+                    gamePath = gamePath.Combine("game");
             }
 
-            yield return new GameLocatorResult(found.Path, fs, GameStore.GOG, new HeroicGOGLocatorResultMetadata
+            yield return new GameLocatorResult(gamePath, fs, GameStore.GOG, new HeroicGOGLocatorResultMetadata
             {
                 Id = id,
                 BuildId = found.BuildId,

--- a/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs
+++ b/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs
@@ -46,9 +46,14 @@ public class HeroicGogLocator : IGameLocator
             if (!_cachedGames.TryGetValue(GOGGameId.From(id), out var found)) continue;
             var fs = found.Path.FileSystem;
 
-            if (found is HeroicGOGGame heroicGOGGame && heroicGOGGame.WinePrefixPath.DirectoryExists())
+            if (found is HeroicGOGGame heroicGOGGame)
             {
-                fs = heroicGOGGame.GetWinePrefix().CreateOverlayFileSystem(fs);
+                var wineData = heroicGOGGame.WineData;
+                if (wineData is not null)
+                {
+                    if (wineData.WinePrefixPath.DirectoryExists())
+                        fs = heroicGOGGame.GetWinePrefix()!.CreateOverlayFileSystem(fs);
+                }
             }
 
             yield return new GameLocatorResult(found.Path, fs, GameStore.GOG, new HeroicGOGLocatorResultMetadata

--- a/src/NexusMods.StandardGameLocators/Services.cs
+++ b/src/NexusMods.StandardGameLocators/Services.cs
@@ -14,6 +14,7 @@ using GameFinder.StoreHandlers.Xbox;
 using GameFinder.Wine;
 using GameFinder.Wine.Bottles;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using NexusMods.Abstractions.GameLocators;
 using NexusMods.Abstractions.GameLocators.Stores.EGS;
 using NexusMods.Abstractions.GameLocators.Stores.GOG;
@@ -103,7 +104,7 @@ public static class Services
             onLinux: () =>
             {
                 services.AddSingleton<AHandler<SteamGame, AppId>>(provider => new SteamHandler(provider.GetRequiredService<IFileSystem>(), registry: null));
-                services.AddSingleton<HeroicGOGHandler>(provider => new HeroicGOGHandler(provider.GetRequiredService<IFileSystem>()));
+                services.AddSingleton<HeroicGOGHandler>(provider => new HeroicGOGHandler(provider.GetRequiredService<IFileSystem>(), provider.GetRequiredService<ILogger<HeroicGogLocator>>()));
 
                 if (registerWine)
                 {

--- a/tests/NexusMods.StandardGameLocators.TestHelpers/Services.cs
+++ b/tests/NexusMods.StandardGameLocators.TestHelpers/Services.cs
@@ -11,6 +11,7 @@ using GameFinder.StoreHandlers.Xbox;
 using GameFinder.Wine;
 using GameFinder.Wine.Bottles;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using NexusMods.Abstractions.Games;
 using NexusMods.Abstractions.Loadouts;
 using NexusMods.Extensions.DependencyInjection;
@@ -158,7 +159,7 @@ public static class Services
 
         if (OSInformation.Shared.IsLinux)
         {
-            coll.AddSingleton<HeroicGOGHandler>(s => new HeroicGOGHandler(s.GetRequiredService<IFileSystem>()));
+            coll.AddSingleton<HeroicGOGHandler>(s => new HeroicGOGHandler(s.GetRequiredService<IFileSystem>(), s.GetRequiredService<ILogger<HeroicGOGHandler>>()));
 
             coll.AddSingleton<IWinePrefixManager<WinePrefix>>(s =>
                 new StubbedWinePrefixManager<WinePrefix>(s.GetRequiredService<TemporaryFileManager>(),

